cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -368,13 +368,16 @@ template <typename IndexIterator>
 size_type validate_segmented_indices(IndexIterator indices_begin, IndexIterator indices_end)
 {
   auto const num_indices = static_cast<size_type>(std::distance(indices_begin, indices_end));
```

Comment:
Should this be using `thrust::distance` instead of `std::distance`? I'm not sure if `std::distance` works on all iterator types. (I could be wrong here.)

Reply:
That's a good question. I think it's worth making the change since I can see some code paths that call through to this with thrust iterators. I'm guessing everything works fine or we'd be seeing bugs, but best to be safe nonetheless.

I'll do this in a separate PR though.